app/Services/VocabularyServiceProvider.js

```diff
@@ -2,41 +2,41 @@ const { VocabularyCard, Translation } = require('../../database');
 const { Drawer } = require('../../database');
 
 // create language package
-async function createVocabularyCard({ languagePackageId, groupId }, name, userId, activate) {
+async function createVocabularyCard({ languagePackageId, groupId }, name, description, userId, activate, res) {
   // if activate = false store vocabulary card in drawer 0 directly
+
+  let drawer;
+
   if (!activate) {
-    const drawer = await Drawer.findOne({
-      attributes: ['id', 'stage'],
+    drawer = await Drawer.findOne({
+      attributes: ['id'],
       where: {
         stage: 0,
         languagePackageId,
         userId,
       },
     });
 
-    const vocabularyCard = await VocabularyCard.create({
-      userId: userId,
-      languagePackageId: languagePackageId,
-      groupId: groupId,
-      drawerId: drawer.id,
-      name: name,
-      lastQuery: new Date(),
-      active: true,
+    if (!drawer) {
+      res.status(400).end();
+      return false;
+    }
+  } else {
+    // if user directly activates card, store it in drawer 1
+    drawer = await Drawer.findOne({
+      attributes: ['id'],
+      where: {
+        stage: 1,
+        languagePackageId: languagePackageId,
+        userId: userId,
+      },
     });
 
-    return vocabularyCard;
+    if (!drawer) {
+      res.status(400).end();
+      return false;
+    }
   }
-
-  // if user directly activates card, store it in drawer 1
-  const drawer = await Drawer.findOne({
-    attributes: ['id', 'stage'],
-    where: {
-      stage: 1,
-      languagePackageId: languagePackageId,
-      userId: userId,
-    },
-  });
-
   // create date the day before yesterday so it will appear in the inbox for querying
   let date = new Date();
```

Comment:
```js
const yesterday = new Date();
yesterday.setDate(yesterday.getDate() - 1);
```
or we create a shift date util function.

Reply:
I don't think we need date shift that much, so we shouldn't need an external function in utils

Reply:
Ok. Then we leave it as it is.